### PR TITLE
feat: updated onetrust_cookie_categories schema structure for amplitude, adobe_analytics and snowflake

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## Description of the change

Updated `onetrust_cookie_categories` schema structure for amplitude, adobe_analytics and snowflake

## Notion Link

> [Notion ticket](https://www.notion.so/rudderstacks/Enabling-oneTrust-Cookie-Categories-for-cloud-mode-b2e01d62ad284e16b95dcd1db6ad6264)

**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by ktgowtham — APPROVED** [review]
Can we update `_test.go` files for affected destinations with oneTrustCookieCategories?

File: rudderstack/integrations/destinations/destination_amplitude.go [hunk 1]
```diff
@@ -472,19 +472,10 @@ func init() {
...
-					Schema: map[string]*schema.Schema{
-						"web": {
-							Type:     schema.TypeList,
-							Optional: true,
-							Elem: &schema.Schema{
-								Type: schema.TypeString,
-							},
-						},
-					},
+				Elem: &schema.Schema{
```
Comment (krishna2020) [inline rudderstack/integrations/destinations/destination_amplitude.go:477]: Shouldn't this be the schema ?
```
"onetrust_cookie_categories": {
				Type:        schema.TypeList,
				Optional:    true,
				Description: "Specify the OneTrust category name for mapping the OneTrust consent settings to RudderStack's consent purposes.",
				Elem: &schema.Schema{
					Type: schema.TypeString,
				},
			}
```

File: rudderstack/integrations/destinations/destination_adobe_analytics.go [hunk 2]
```diff
@@ -582,19 +582,10 @@ func init() {
...
-					Schema: map[string]*schema.Schema{
-						"web": {
-							Type:     schema.TypeList,
-							Optional: true,
-							Elem: &schema.Schema{
-								Type: schema.TypeString,
-							},
-						},
-					},
+				Elem: &schema.Schema{
```
Comment (krishna2020) [inline rudderstack/integrations/destinations/destination_adobe_analytics.go:587]: can you re-check this schema too

File: rudderstack/integrations/destinations/destination_amplitude.go (hunk 1 above)
Comment (krishna2020) [reply rudderstack/integrations/destinations/destination_amplitude.go:477]: Ignore the comment - it's right

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**